pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -10,36 +10,29 @@ import (
 	"os"
 	"strings"
 
-	zconfig "github.com/lf-edge/eve/api/go/config"
 	"github.com/lf-edge/eve/pkg/pillar/containerd"
 	"github.com/lf-edge/eve/pkg/pillar/diskmetrics"
 	"github.com/lf-edge/eve/pkg/pillar/types"
-	"github.com/lf-edge/eve/pkg/pillar/utils"
 	log "github.com/sirupsen/logrus"
 )
 
 // createVolume does not update status but returns
 // new values for VolumeCreated, FileLocation, and error
-func createVolume(ctx *volumemgrContext, status types.OldVolumeStatus) (bool, string, error) {
+func createVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool, string, error) {
 
 	srcLocation := status.FileLocation
 	log.Infof("createVolume(%s) from %s", status.Key(), srcLocation)
-	switch status.Origin {
```

Comment:
Don't you need to keep the check on Origin?

Reply:
What is the use of Origin? What is it for?

Reply:
For now it is just the distinction between something to be downloaded and a blank volume. But if status.Origin is not "download", then the DownloaderOriginConfig pointer is NULL.

Didn't you copy this code from the original create.go file? It had that check.

Reply:
The logic of downloaded vs blank volume is implemented in new doUpdateVol function.

Reply:
Even so it is better to have the log.Fatal here then tripping on a null point if this ever gets called for blank.

Reply:
Not needed